libs/net/elastic/TNOElasticClient.cs

```diff
@@ -0,0 +1,85 @@
+        #endregion
+
+        #region Constructors
+        /// <summary>
+        /// Creates a new instance of a TNOElasticClient object, initializes with specified parameters.
+        /// </summary>
+        /// <param name="httpClient"></param>
+        /// <param name="options"></param>
+        public TNOElasticClient(IHttpRequestClient httpClient, IOptions<ElasticOptions> options) : base(GetConnectionSettings(options.Value))
+        {
+            this.Client = httpClient;
+            this.Options = options.Value;
+            if (this.Options.Url == null) throw new ConfigurationException("Elastic Url configuration is required.");
+            var username = Environment.GetEnvironmentVariable("ELASTIC_USERNAME") ?? throw new ConfigurationException("Elastic environment variable 'ELASTIC_USERNAME' is required.");
+            var password = Environment.GetEnvironmentVariable("ELASTIC_PASSWORD") ?? throw new ConfigurationException("Elastic environment variable 'ELASTIC_PASSWORD' is required.");
+            this.Credentials = new BasicAuthenticationCredentials(username, password);
+            var credentials = Convert.ToBase64String(System.Text.ASCIIEncoding.ASCII.GetBytes($"{this.Credentials.Username}:{password}"));
+            this.Client.Client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", credentials);
+        }
+        #endregion
+
+        #region Methods
+        /// <summary>
+        /// Get the connection string for elasticsearch.
+        /// </summary>
+        /// <param name="config"></param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        private static ConnectionSettings GetConnectionSettings(ElasticOptions options)
+        {
+            if (options.Url == null) throw new ConfigurationException("Elastic configuration property 'Elastic:Url' is required'");
+            var username = Environment.GetEnvironmentVariable("ELASTIC_USERNAME") ?? throw new ConfigurationException("Elastic environment variable 'ELASTIC_USERNAME' is required.");
+            var password = Environment.GetEnvironmentVariable("ELASTIC_PASSWORD") ?? throw new ConfigurationException("Elastic environment variable 'ELASTIC_PASSWORD' is required.");
+
+            return new ConnectionSettings(options.Url)
+                .BasicAuthentication(username, password)
+                .DefaultIndex(options.UnpublishedIndex)
+                .EnableApiVersioningHeader()
+                .RequestTimeout(new TimeSpan(0, 30, 0))
+                .ThrowExceptions();
+        }
+
+        /// <summary>
+        /// Make a request to Elasticsearch 'index' with the specified 'query'.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="index"></param>
+        /// <param name="query"></param>
+        /// <returns></returns>
+        public async Task<SearchResultModel<T>> SearchAsync<T>(string index, JsonDocument query)
```

Comment:
This is the magic.  We can now submit raw JSON queries to Elasticsearch.  No longer do we need to hop through hoops with NEST.  I honestly don't understand why NEST is built to be so difficult to use.